code/FulpstationCode/sec_expanded_loadouts/detective_goggles.dm

```diff
@@ -1,169 +1,169 @@
+
+/obj/item/clothing/glasses/detective
+	name = "detective's glasses"
+	desc = "Stylish glasses with integrated medical, diagnostic and security HUDs and reagent scanning used by detectives. The Meson Scanner mode lets you see basic structural and terrain layouts through walls. WARNING! Flash proofing has been compromised to accomodate HUD integration."
+	icon = 'icons/Fulpicons/Surreal_stuff/detective_obs.dmi'
+	worn_icon = 'icons/Fulpicons/Surreal_stuff/detective_obs_worn.dmi'
+	icon_state = "sundetect-"
+	inhand_icon_state = "sunglasses"
+	actions_types = list(/datum/action/item_action/toggle_mode)
+
+	vision_flags = NONE
+	darkness_view = 2
+	invis_view = SEE_INVISIBLE_LIVING
+
+	var/list/modes = list(MODE_NONE = MODE_MESON, MODE_MESON = MODE_NONE)
+	var/mode = MODE_NONE
+	var/range = 1
+	var/emped = FALSE //whether or not it's subject to the effects of an EMP.
+
+	clothing_flags = SCAN_REAGENTS //You can see reagents while wearing detective glasses
+	resistance_flags = ACID_PROOF
+	glass_colour_type = /datum/client_colour/glass_colour/red
+	armor = list("melee" = 0, "bullet" = 0, "laser" = 5, "energy" = 0, "bomb" = 0, "bio" = 0, "rad" = 25, "fire" = 100, "acid" = 100)
+
+/obj/item/clothing/glasses/detective/Initialize()
+	. = ..()
+	update_icon()
+
+/obj/item/clothing/glasses/detective/dropped(mob/user)
+	..()
+	remove_sensors(user)
+
+/obj/item/clothing/glasses/detective/equipped(mob/user, slot)
+	..()
+	add_sensors(user, slot)
+
+/obj/item/clothing/glasses/detective/proc/remove_sensors(mob/user)
+	if(!user)
+		if(ismob(loc))
+			user = loc
+		else
+			return
+	var/datum/atom_hud/secsensor = GLOB.huds[DATA_HUD_SECURITY_ADVANCED]
+	var/datum/atom_hud/medsensor = GLOB.huds[DATA_HUD_MEDICAL_ADVANCED]
+	var/datum/atom_hud/diagsensor = GLOB.huds[DATA_HUD_DIAGNOSTIC_ADVANCED]
+	secsensor.remove_hud_from(user)
+	medsensor.remove_hud_from(user)
+	diagsensor.remove_hud_from(user)
+
+/obj/item/clothing/glasses/detective/proc/add_sensors(mob/user, slot)
+	if(emped) //doesn't function while affected by EMPs.
+		return
+	if(slot != ITEM_SLOT_EYES)
+		return
+	if(!user)
+		if(ismob(loc))
+			user = loc
+		else
+			return
+	var/datum/atom_hud/secsensor = GLOB.huds[DATA_HUD_SECURITY_ADVANCED]
+	var/datum/atom_hud/medsensor = GLOB.huds[DATA_HUD_MEDICAL_ADVANCED]
+	var/datum/atom_hud/diagsensor = GLOB.huds[DATA_HUD_DIAGNOSTIC_ADVANCED]
+	secsensor.add_hud_to(user)
+	medsensor.add_hud_to(user)
+	diagsensor.add_hud_to(user)
+
+
+/obj/item/clothing/glasses/detective/emp_act(severity)
+	. = ..()
+	if(. & EMP_PROTECT_SELF)
+		return
+	emp_overload(severity)
+
+/obj/item/clothing/glasses/detective/proc/emp_overload(severity)
+	if(ismob(src.loc))
+		var/mob/M = src.loc
+		to_chat(M, "<span class='danger'>[src]' hud abruptly flickers out as it overloads!</span>")
+		remove_sensors(M)
+	if(mode == MODE_MESON) //disable mesons if active
+		vision_flags = NONE
+		darkness_view = 2
+		lighting_alpha = null
+		mode = MODE_NONE
```

Comment:
The whole file changing is a result of a line endings thing. I didn't think it was important, but I can resolve it if you'd like. 

In any case, someone copied `if(MODE_MESON)` over here from the switch statement. Thank byond for not having `case :`